lib/tapioca/commands/annotations.rb

```diff
@@ -60,10 +60,19 @@ def remove_expired_annotations(project_gems)
         say("\nDone\n\n", :green)
       end
 
-      sig { returns(RepoIndex) }
-      def fetch_index
-        say("Retrieving index from central repository... ", [:blue, :bold])
-        content = fetch_file(CENTRAL_REPO_INDEX_PATH)
+      sig { returns(T::Hash[String, RepoIndex]) }
+      def fetch_indexes
+        repo_number = 1
+        @central_repo_root_uris.each_with_object({}) do |uri, hash|
+          hash[uri] = fetch_index(uri, repo_number: @central_repo_root_uris.size > 1 ? repo_number : nil)
```

Comment:
`@central_repo_root_uris.size > 1` can probably be a local? It doesn't need to be called every time.